### PR TITLE
CI for Loudly

### PR DESCRIPTION
**Pull request by ZeRoGerc**
- Robolectric test for MainActivity
 - Travis integration

### CODE REVIEW

File: app/src/main/java/ly/loud/loudly/test_utils/FacebookClientTestImpl.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import retrofit2.http.Part;
+import retrofit2.http.Path;
+import retrofit2.http.Query;
+import retrofit2.http.Url;
+
+public class FacebookClientTestImpl implements FacebookClient {
+
+    @NonNull
+    @Override
+    public Call<Data<List<Post>>> loadPosts(@Nullable @Query("since") Long since, @Nullable @Query("until") Long until, @NonNull @Query("access_token") String accessToken) {
```
Comment (danilkolikov) [inline app/src/main/java/ly/loud/loudly/test_utils/FacebookClientTestImpl.java]: Split to multiple lines

File: app/src/main/java/ly/loud/loudly/test_utils/FacebookClientTestImpl.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        return null;
```
Comment (danilkolikov) [inline app/src/main/java/ly/loud/loudly/test_utils/FacebookClientTestImpl.java]: Return null from NonNull?

File: app/src/main/java/ly/loud/loudly/test_utils/FacebookClientTestImpl.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
+package ly.loud.loudly.test_utils;
```
Comment (danilkolikov) [inline app/src/main/java/ly/loud/loudly/test_utils/FacebookClientTestImpl.java]: I suppose now there is a lot of garbage in test_utils

File: app/src/test/java/ly/loud/loudly/application/TempActivityTest.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+import ly.loud.loudly.BuildConfig;
+import ly.loud.loudly.test_utils.LoudlyIntegrationTestRunner;
+
+@RunWith(LoudlyIntegrationTestRunner.class)
+@Config(constants = BuildConfig.class, sdk = 21)
+public class TempActivityTest {
+    @Test
+    public void onCreate_shouldWorkOnNormalLifecycle() {
+        final ActivityController<TempActivity> controller = Robolectric.buildActivity(TempActivity.class);
```
Comment (danilkolikov) [inline app/src/test/java/ly/loud/loudly/application/TempActivityTest.java]: static import?

File: app/src/test/java/ly/loud/loudly/test_utils/TestContext.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    public static ShadowApplication shapp;
+
+    /**
+     * Should be called before each test
+     */
+    public static void init() {
+        /*
+            Application is recreated for every test, so we need to reinitialize these
+         */
+//        shapp = Shadows.shadowOf(LoudlyIntegrationTestRunner.app());
```
Comment (danilkolikov) [inline app/src/test/java/ly/loud/loudly/test_utils/TestContext.java]: Commented?

File: app/src/main/java/ly/loud/loudly/application/TempActivity.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package ly.loud.loudly.application;
+
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+
+import ly.loud.loudly.R;
+
+public class TempActivity extends AppCompatActivity {
```
Comment (danilkolikov) [inline app/src/main/java/ly/loud/loudly/application/TempActivity.java]: Why do we need temp activity now, if we have tests for MainActivity

File: app/src/main/java/ly/loud/loudly/test_utils/FacebookClientTestImpl.java (hunk 2 above)
Comment (ZeRoGerc) [inline app/src/main/java/ly/loud/loudly/test_utils/FacebookClientTestImpl.java]: will be implemented as soon as we want to test interactions with networks

File: app/src/test/java/ly/loud/loudly/application/TempActivityTest.java (hunk 4 above)
Comment (ZeRoGerc) [inline app/src/test/java/ly/loud/loudly/application/TempActivityTest.java]: maybe

File: app/src/main/java/ly/loud/loudly/test_utils/FacebookClientTestImpl.java (hunk 3 above)
Comment (ZeRoGerc) [inline app/src/main/java/ly/loud/loudly/test_utils/FacebookClientTestImpl.java]: fixed

**Comment by danilkolikov** [comment]
Ok if it works

**Pull request merged**